### PR TITLE
Rank search results in the database

### PR DESCRIPTION
**Pull request by crisptrutski**
A big gross, but getting the job done for now.

### CODE REVIEW

File: src/metabase/models.clj [hunk 1]
```diff
@@ -54,9 +56,12 @@
    [metabase.models.view-log :as view-log]
    [metabase.plugins.classloader :as classloader]
    [metabase.public-settings.premium-features :refer [defenterprise]]
+   [metabase.search :as search]
+   [metabase.search.postgres.ingestion :as search.ingestion]
```
Comment (piranha) [inline src/metabase/models.clj:60]: well it should be something like `search/update-index!`, and then it'll look at what's the current engine and proceed, right?

File: src/metabase/search/postgres/index.clj [hunk 2]
```diff
@@ -74,12 +75,15 @@
...
             table-name (name pending-table)]
-        (t2/query
-         (format "CREATE UNIQUE INDEX IF NOT EXISTS %s_identity_idx ON %s (model, model_id)" idx_prefix table-name))
-        (t2/query
-         (format "CREATE INDEX IF NOT EXISTS %s_tsvector_idx ON %s USING gin (search_vector)" idx_prefix table-name))))
+        (doseq [stmt ["CREATE UNIQUE INDEX IF NOT EXISTS %s_identity_idx ON %s (model, model_id)"
+                      "CREATE INDEX IF NOT EXISTS %s_tsvector_idx ON %s USING gin (search_vector)"
+                      ;; Spam all the indexes for now, let's see if they get used on Stats / Ephemeral envs.
+                      "CREATE INDEX IF NOT EXISTS %s_model_idx ON %s (model)"
+                      "CREATE INDEX IF NOT EXISTS %s_model_archived_idx ON %s (model, archived)"
```
Comment (piranha) [inline src/metabase/search/postgres/index.clj:84]: those two (`model` and `model + archived`) are almost the same, at least removing first will make postgres use the second.

**Review by sansolovyov — COMMENTED** [review]
👌

File: src/metabase/search/postgres/index.clj (hunk 2 above)
Comment (crisptrutski) [reply src/metabase/search/postgres/index.clj:84]: Yeah I added models first, then the hybrid when I realized it wasn't helping.

Didn't realize the original was actively causing trouble though 😠

**Review by sansolovyov — APPROVED** [review]

**Pull request merged**